dlib/dnn/layers.h

```diff
@@ -739,6 +739,119 @@ namespace dlib
         >
     using upsample = add_layer<upsample_<scale,scale>, SUBNET>;
 
+// ----------------------------------------------------------------------------------------
+
+    template <
+        long NR_, 
+        long NC_
+        >
+    class resize_to_
+    {
+    public:
+        static_assert(NR_ >= 1, "NR resize parameter can't be less than 1.");
+        static_assert(NC_ >= 1, "NC resize parameter can't be less than 1.");
+        
+        resize_to_()
+        {
+        }
+        
+        template <typename SUBNET>
+        void setup (const SUBNET& sub)
+        {
+            scale_x = (double)NC_/(double)sub.get_output().nc();
```

Comment:
This should go in the forward() function since the size of the network output could change from one invocation to the next.  Some layers require the input size to be constant, like the fc_ layer and they make more heavy use of setup() as a result.  But layers that make sense for tensors of varying sizes should support that.  So here that means these scale parameters should be calculated each time at the start of forward().